pyaedt/modeler/cad/Primitives3D.py

```diff
@@ -3339,3 +3339,96 @@ def _check_value_type(self, taken_value, value_type, are_inequations_checkable,
+            - if ``geo_type=='cylinder'``, expected fields are ``'xc'`` (base
+            center x coordinate),``'yc'`` (base center y coordinate), ``'zc'``
+            (base center z coordinate), ``'height'``, ``'radius'``,
+            ``'iradius'`` (inner radius for hollow cylinder), ``'plane'``
+            (plane on which the cylinder base lies).
+        header_line : int, optional
+            The line marking the start of data.
+            The default value is ``2``.
+        column_mapping : dict, optional
+            Map of the csv columns names to the accepted column names.
+            The default value is ``None``.
+        geo_type: str, optional
+            Type of geometry that the csv file describe. Accepted value is: ``"cylinder"``.
+            Default is ``"cylinder"``.
+        unit : str , optional
+            Unit of values used to create geometry. Default unit is ``"mm"``.
+        Returns
+        -------
+        list of :class:`pyaedt.modeler.cad.object3d.Object3d`.
+        """
+        import pandas as pd
+
+        df = pd.read_csv(csv_file, header=header_line)
+        if column_mapping is not None:
+            df = df.rename(columns=column_mapping)
+        if geo_type == "cylinder":
+            return self._create_cylinder(df, unit=unit)
+        else:
+            raise ValueError("Geometry type not supported yet.")
+
+    @pyaedt_function_handler()
+    def _remove_special_characters(self, text):
+        import re
+
+        sanitized_text = re.sub("[^a-zA-Z0-9_ \n]", "_", text)
+        self.logger.info("Renamed " + text + " to " + sanitized_text)
+        return sanitized_text
+
+    @pyaedt_function_handler()
+    def _length_unit_conversion(self, value, input_units):
+        from pyaedt.generic.constants import unit_converter
+
+        converted_value = unit_converter(value, unit_system="Length", input_units=input_units, output_units="mm")
+        return converted_value
+
+    @pyaedt_function_handler()
+    def _create_cylinder(self, df, unit):
+        cylinders = []
+        for i in range(len(df)):
+            try:
+                name = self._remove_special_characters(df["name"][i])
+                radius = self._length_unit_conversion(df["radius"][i], input_units=unit)
+                height = self._length_unit_conversion(df["height"][i], input_units=unit)
+                cs_axis = self._length_unit_conversion(df["plane"][i], input_units=unit)
+                inner_radius = self._length_unit_conversion(df["iradius"][i], input_units=unit)
+                xc = self._length_unit_conversion(df["xc"][i], input_units=unit)
+                yc = self._length_unit_conversion(df["yc"][i], input_units=unit)
+                zc = self._length_unit_conversion(df["zc"][i], input_units=unit)
+                position = (xc, yc, zc)
+            except KeyError:
+                self.logger.info("The column names in the file do not match the expected names")
```

Comment:
Error messages should have concluding punction, which is almost always a period (.).